Add support for building xkbcommon and xcb to build_visit.

brugger1 opened the pull request:
### Description

I added support for building `ninja`, `meson`, `xkbcommon` and `xcb` to `build_visit`. This will allow building VisIt with Qt6 on systems without the `xkbcommon` and `xcb` development packages installed, which appears to be most supercomputer systems these days.

The goal was to add support for building `xkbcommon` and `xcb`, but `xkbcommon` uses `meson`, which uses `ninja`, so all four were added to `build_visit`. They are enabled with:
```
build_visit --ninja --meson --xkbcommon --xcb`
```

I am putting this on the 3.4RC so that we can use `build_visit` to build 3.4.2 on some additional systems. When I merge the changes to develop I will add release notes. I will also update the build scripts and config site files for building on `sierra` and its associated systems on another pull request.

### Type of change

* [X] New feature~~

### How Has This Been Tested?

I tested `build_visit` on `rzansel`, which doesn't have the `xkbcommon` and `xcb` development packages installed. I invoked `build_visit` with:
```
./build_visit3_4_2 --required --parallel --no-visit --qwt --cmake --ninja --zlib --python --meson --xkbcommon --xcb --llvm --mesagl --qt6 --silo --hdf5 --makeflags -j8
```

I installed the resulting distribution and ran the VisIt GUI. The Qt6 GUI came up and I created pseudocolor and mesh plots from `curv2d.silo` and saved the resulting image.

### Reminders:

- Please follow the [style guidelines][1] of this project.
- Please perform a self-review of your code before submitting a PR and asking others to review it.
- Please assign reviewers (see [VisIt's PR procedures][2] for more information).

### Checklist:

- [X] I have commented my code where applicable.~~
- ~~[ ] I have updated the release notes.~~
- ~~[ ] I have made corresponding changes to the documentation.~~
- ~~[ ] I have added debugging support to my changes.~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works.~~
- ~~[ ] I have confirmed new and existing unit tests pass locally with my changes.~~
- ~~[ ] I have added new baselines for any new tests to the repo.~~
- [X] I have NOT made any changes to [*protocol* or *public interfaces*][3] in an RC branch.~~

[1]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/StyleGuide.html
[2]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/pr_create.html#reviewers
[3]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/RCDevelopment.html#communication-protocols-and-public-apis

Commit:
Add support for building ninja to build_visit.

Commit:
Add ability to optionally install meson as part of python.

Commit:
Fix an bug setting the install directory with building ninja.

Commit:
Remove installing of meson package from bv_python.

Commit:
Add support for installing meson to build_visit.

Commit:
Fix a typo in a comment.

Commit:
Make meson script executable.

Commit:
Fix ninja install directory.

Commit:
Add support for building xkbcommon to build_visit.

Commit:
Fix a bug in bv_meson.sh.

Commit:
Code cleanup in bv_meson.sh.

Commit:
Add ability to output pkgconfig files to bv_xkbcommon.h.

Commit:
Cleanup the code in bv_xkbcommon.sh.

Commit:
Fix a typo in bv_xkbcommon.sh.

Commit:
Connect xkbcommon to qt6 in build_visit.

Commit:
Fix a typo in bv_xkbcommon.sh.

Commit:
Fix a typo in bv_xkbcommon.sh.

Commit:
Add support for building xcb to build_visit.

Commit:
Make qt6 depend on xcb if building xcb.

Commit:
Fix the install path for xcb.

Commit:
Fix a couple of bugs building xcb.

Commit:
Modify bv_qt6 to handle case where xcb already built.

Commit:
A few improvements to building xkbcommon.

Commit:
Remove some redundant code.

Commit:
Add print xcb verion to bv_xcb.sh.

biagas commented on src/tools/dev/scripts/bv_support/bv_ninja.sh:
```diff
@@ -0,0 +1,184 @@
+    vopts="${vopts} -DCMAKE_BUILD_TYPE:STRING=${VISIT_BUILD_MODE}"
+
+    #
+    # Several platforms have had problems with the VTK cmake configure
```
```suggestion
    # Several platforms have had problems with the cmake configure
```
But do we really need this comment any more?

brugger1 replied:
No, we probably don't need the comment anymore. I removed it.

biagas reviewed (commented):
Does xcb/xkbcommon need to be installed alongside VisIt if they aren't part of the system. If so, is that coming with a different PR?

Commit:
Remove unnecessary comment.

brugger1 commented:
> Does xcb/xkbcommon need to be installed alongside VisIt if they aren't part of the system. If so, is that coming with a different PR?

I would image so, but it didn't seem necessary. I renamed the `third_party` directory so that VisIt couldn't find the libraries it linked against and then ran VisIt and it ran fine. My guess is that even though the development packages aren't installed, the runtime packages are, so they don't need to be duplicated.

I can add that in another PR.

Commit:
Add info to host profile for xkbcommon and xcb.

Commit:
Add xcb xkbcommon libraries to the distribution.

brugger1 commented:
I added logic to copy the `xcb` and `xkbcommon` libraries to the lib directory when doing development and also added them to the install so that they are included in a distribution.

brugger1 commented:
Here is the result from testing the packaging of the `xcb` and `xkbcommon` libraries on `rzansel`.
```
ls -l visit/3.4.2/linux-ppc64le/lib/libxcb*
lrwxrwxrwx 1 brugger brugger     20 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxcb-ewmh.so -> libxcb-ewmh.so.2.0.0
lrwxrwxrwx 1 brugger brugger     20 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxcb-ewmh.so.2 -> libxcb-ewmh.so.2.0.0
-rw-r--r-- 1 brugger brugger 191512 Jan 21 09:05 visit/3.4.2/linux-ppc64le/lib/libxcb-ewmh.so.2.0.0
lrwxrwxrwx 1 brugger brugger     21 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxcb-icccm.so -> libxcb-icccm.so.4.0.0
lrwxrwxrwx 1 brugger brugger     21 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxcb-icccm.so.4 -> libxcb-icccm.so.4.0.0
-rw-r--r-- 1 brugger brugger 113560 Jan 21 09:05 visit/3.4.2/linux-ppc64le/lib/libxcb-icccm.so.4.0.0
lrwxrwxrwx 1 brugger brugger     21 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxcb-image.so -> libxcb-image.so.0.0.0
lrwxrwxrwx 1 brugger brugger     21 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxcb-image.so.0 -> libxcb-image.so.0.0.0
-rw-r--r-- 1 brugger brugger 116464 Jan 21 09:04 visit/3.4.2/linux-ppc64le/lib/libxcb-image.so.0.0.0
lrwxrwxrwx 1 brugger brugger     23 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxcb-keysyms.so -> libxcb-keysyms.so.1.0.0
lrwxrwxrwx 1 brugger brugger     23 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxcb-keysyms.so.1 -> libxcb-keysyms.so.1.0.0
-rw-r--r-- 1 brugger brugger  84296 Jan 21 09:04 visit/3.4.2/linux-ppc64le/lib/libxcb-keysyms.so.1.0.0
lrwxrwxrwx 1 brugger brugger     27 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxcb-render-util.so -> libxcb-render-util.so.0.0.0
lrwxrwxrwx 1 brugger brugger     27 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxcb-render-util.so.0 -> libxcb-render-util.so.0.0.0
-rw-r--r-- 1 brugger brugger  98176 Jan 21 09:06 visit/3.4.2/linux-ppc64le/lib/libxcb-render-util.so.0.0.0
lrwxrwxrwx 1 brugger brugger     20 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxcb-util.so -> libxcb-util.so.1.0.0
lrwxrwxrwx 1 brugger brugger     20 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxcb-util.so.1 -> libxcb-util.so.1.0.0
-rw-r--r-- 1 brugger brugger 103008 Jan 21 09:03 visit/3.4.2/linux-ppc64le/lib/libxcb-util.so.1.0.0
rzansel61{brugger}783: ls -l visit/3.4.2/linux-ppc64le/lib/libxkb*
lrwxrwxrwx 1 brugger brugger     25 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxkbcommon-x11.so -> libxkbcommon-x11.so.0.0.0
lrwxrwxrwx 1 brugger brugger     25 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxkbcommon-x11.so.0 -> libxkbcommon-x11.so.0.0.0
-rw-r--r-- 1 brugger brugger 141368 Jan 21 09:03 visit/3.4.2/linux-ppc64le/lib/libxkbcommon-x11.so.0.0.0
lrwxrwxrwx 1 brugger brugger     21 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxkbcommon.so -> libxkbcommon.so.0.0.0
lrwxrwxrwx 1 brugger brugger     21 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxkbcommon.so.0 -> libxkbcommon.so.0.0.0
-rw-r--r-- 1 brugger brugger 694504 Jan 21 09:03 visit/3.4.2/linux-ppc64le/lib/libxkbcommon.so.0.0.0
lrwxrwxrwx 1 brugger brugger     23 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxkbregistry.so -> libxkbregistry.so.0.0.0
lrwxrwxrwx 1 brugger brugger     23 Jan 23 13:54 visit/3.4.2/linux-ppc64le/lib/libxkbregistry.so.0 -> libxkbregistry.so.0.0.0
-rw-r--r-- 1 brugger brugger 110608 Jan 21 09:03 visit/3.4.2/linux-ppc64le/lib/libxkbregistry.so.0.0.0
```
